sched/jobDefSerializer_test.go

```diff
@@ -0,0 +1,249 @@
+		}
+		if def1Task.SnapshotId != def2Task.SnapshotId {
+			return false, fmt.Sprintf("Snapshot ids differ, expected %s, got %s", def1Task.SnapshotId, def2Task.SnapshotId)
+		}
+	}
+
+	return true, ""
+}
+
+func stringMapEqual(expectedMap, actualMap map[string]string) (bool, string) {
+	if len(expectedMap) != len(actualMap) {
+		return false, fmt.Sprintf("The 2 maps are different lengths: expected %d, got %d", len(expectedMap), len(actualMap))
+	}
+
+	for taskName, entry1 := range expectedMap {
+		if entry2, ok := actualMap[taskName]; !ok || entry1 != entry2 {
+			return false, fmt.Sprintf("The map entries (expected %s, got %s) for %s have different values.", expectedMap[taskName], actualMap[taskName], taskName)
+		}
+	}
+
+	return true, ""
+}
+
+func stringSliceCompare(expectedSlice, actualSlice []string) (bool, string) {
+	if len(expectedSlice) != len(actualSlice) {
+		return false, fmt.Sprintf("The 2 slices (expected %s, got %s) are different lengths: expected %d, got %d", expectedSlice, actualSlice, len(expectedSlice), len(actualSlice))
+	}
+
+	for j, val1 := range expectedSlice {
+		if actualSlice[j] != val1 {
+			return false, fmt.Sprintf("Slice entries differ at index %d: expected %s, got %s", j, expectedSlice[j], actualSlice[j])
+		}
+	}
+	return true, ""
+}
+
+func Print(jobDef JobDefinition) {
+	for taskName, taskDef := range jobDef.Tasks {
+		fmt.Printf(fmt.Sprintf("taskName: %s\n", taskName))
+		fmt.Printf(fmt.Sprintf("\ttimeout: %s\n", taskDef.Timeout.String()))
+		fmt.Printf(fmt.Sprintf("\tsnapshotId: %s\n", taskDef.SnapshotId))
+		for envVarName, envVarVal := range taskDef.EnvVars {
+			fmt.Printf(fmt.Sprintf("\tenvVar:%s = %s\n", envVarName, envVarVal))
+		}
+		for i, arg := range taskDef.Argv {
+			fmt.Printf(fmt.Sprintf("\targ[%d]:%s\n", i, arg))
+		}
+	}
+	fmt.Printf("\n")
+}
+
+func Test_RandomSerializerDeserializer(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+	parameters.MinSuccessfulTests = 1000
+	properties := gopter.NewProperties(parameters)
+
+	properties.Property("Serialize JobDef", prop.ForAll(
+		func(jobDef JobDefinition) bool {
+			//serialize then deserialized -  should get equal objects
+			if defAsByte, err := Serialize(&jobDef, serializeType); err != nil {
+				return false
+			} else {
+				deserializedJobDef, err := Deserialize(defAsByte, serializeType)
+				if err != nil {
+					return false
+				}
+				if ok, err1 := jobDef.Equal(*deserializedJobDef); !ok {
+					fmt.Printf(err1)
+					return false
+				}
+				//fmt.Printf(fmt.Sprintf("Job Def:%s\n",string(defAsByte)))
+				return true;
+			}
+
+		},
+
+		genJobDef(),
+	))
+
+	properties.TestingRun(t)
+
+}
+
+func genJobDef() gopter.Gen {
+	return func(genParams *gopter.GenParameters) *gopter.GenResult {
+		jobDef := genJobDefFromParams(genParams)
+		//Print(jobDef)
+		genResult := gopter.NewGenResult(jobDef, gopter.NoShrinker)
+		return genResult
+	}
+}
+
+func genJobDefFromParams(genParams *gopter.GenParameters) JobDefinition {
```

Comment:
Can you move this much better random JobDef function into scoot/sched/generators.go 

That would help make that code better overall!